### DISCUSSION TITLE
$fetch and useFetch SSR error with proxy request

### DISCUSSION THREAD

**Discussion by palaxius** [nuxt/nuxt#16107] [Questions]
Nuxt 3. When ssr is enabled, an error occurs when using a proxy request on the server side.

**app.vue:**
```
<script setup>
const fetchResponse = ref(null);

// Working as intended with full url
// fetchResponse.value = await $fetch('https://dog.ceo/api/breeds/image/random');

// Working only on client side with proxy url (check nuxt.config.ts)
  fetchResponse.value = await $fetch('test/breeds/image/random');
</script>
```

**nuxt.config.ts:**
```
export default defineNuxtConfig({
  ssr: true,
  nitro: {
    devProxy: {
      '/test': {
        target: 'https://dog.ceo/api',
        changeOrigin: true,
        cookieDomainRewrite: '',
        secure: false,
      },
    },
  },
});
```

Error: Error: Invalid URL () at async setup (./app.js:33:12)

Reproduction: https://stackblitz.com/edit/github-8lgefy-kdefyk?file=app.vue,nuxt.config.ts

How do I send server-side requests that are proxied via nuxt.config ?

**Comment by motea927** [comment]
Hi, this issue is `$fetch` think this URL can be served from `Nitro`,
But Nitro doesn't have this route, you can use the full URL to prevent this error, as below,
```ts
fetchResponse.value = await $fetch(
  'https://localhost:3000/test/breeds/image/random'
);
```
But your `/test` is empty route, so in production you will occur error,
maybe you need create a server/routes file to handle request (dev/prod).
